db/manager.go

```diff
@@ -97,19 +90,19 @@ func NewManager(network net.Net, opts ...Option) (*Manager, error) {
 }
 
 // NewDB creates a new db and prefixes its datastore with base key.
-func (m *Manager) NewDB(ctx context.Context, creds thread.Credentials, collections ...CollectionConfig) (*DB, error) {
-	id := creds.ThreadID()
+func (m *Manager) NewDB(ctx context.Context, id thread.ID, opts ...NewManagedDBOption) (*DB, error) {
```

Comment:
I'd actually love to see the `id thread.ID` param moved to options. If you specify one as an option it's used, otherwise we'll generate a thread id for you.

Reply:
I was thinking the same, but on the client side (for this DB API) once we have client-side ACL builders that involve packing the ACL hash into the ID.

Reply:
ok. maybe we can explore client side sugar for this in the future

Reply:
Totally